From 923224e5a60622736b4754e4bd7cdb0a06513c4f Mon Sep 17 00:00:00 2001
From: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Date: Tue, 2 May 2017 17:13:46 -0700
Subject: [PATCH 0835/2278] usb: dwc3: gadget: Set gadget handle to the phy

This patch adds support to set gadget handle to phy explictly.
Some phy drivers uses this as an indication to change the
OTG state machine to device mode.

Change-Id: I44839f2707b4d52191774d36095233751f4d826f
Signed-off-by: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
---
 drivers/usb/dwc3/gadget.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index b8704c0..ad31f23 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -1925,6 +1925,15 @@ static int dwc3_gadget_start(struct usb_gadget *g,
 	int			ret = 0;
 	int			irq;
 
+	if (dwc->usb2_phy) {
+		ret = otg_set_peripheral(dwc->usb2_phy->otg, &dwc->gadget);
+		if (ret == -ENOTSUPP)
+			dev_info(dwc->dev, "no OTG driver registered\n");
+		else if (ret)
+			return ret;
+	}
+
+
 	irq = dwc->irq_gadget;
 	ret = request_threaded_irq(irq, dwc3_interrupt, dwc3_thread_interrupt,
 			IRQF_SHARED, "dwc3", dwc->ev_buf);
@@ -1973,6 +1982,10 @@ static int dwc3_gadget_stop(struct usb_gadget *g)
 	unsigned long		flags;
 	int			epnum;
 
+
+        if (dwc->usb2_phy)
+                otg_set_peripheral(dwc->usb2_phy->otg, NULL);
+
 	spin_lock_irqsave(&dwc->lock, flags);
 
 	if (pm_runtime_suspended(dwc->dev))
-- 
2.7.4

